subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/TaskCacheabilityReasonIntegrationTest.groovy

```diff
@@ -108,6 +120,26 @@ class TaskCacheabilityReasonIntegrationTest extends AbstractIntegrationSpec impl
         assertCachingDisabledFor null, null
     }
 
+    def "cacheability for a non-cacheable task is NOT_ENABLED_FOR_TASK"() {
+        buildFile << """
+            task notCacheableByDefault(type: NotCacheableByDefault) {}
+        """
+        when:
+        withBuildCache().run "notCacheableByDefault"
+        then:
+        assertCachingDisabledFor NOT_ENABLED_FOR_TASK, "Caching has been disabled for the task"
```

Comment:
I don't think re-using `NOT_ENABLED_FOR_TASK` is a good idea, since then you can't distinguish in build scans between tasks which are not cacheable by choice and tasks which are not cacheable since nobody look at them. And the point is that folks can easily distinguish between the two things, right?

See for example: https://ge.gradle.org/s/gkvo4yzj7jr6k/timeline?cacheability=output_caching_not_enabled

Reply:
As we discussed: the immediate goal is to allow a more detailed description to be presented in build scans and the `INFO` log. Filtering for these tasks specifically is another topic. I added https://github.com/gradle/gradle/issues/16381 to follow up.